src/builtin_cd.c

```diff
@@ -0,0 +1,44 @@
+#include "bash_ops.h"
+#include "env_access.h"
+#include "directory.h"
+#include "ft_printf.h"
+#include <stdlib.h>
+
+int 	go_to_prev_dir()
+{
+	return (directory_change_dir(env_get_prev_dir()));
+}
+
+int	go_to_home()
+{
+	return (directory_change_dir(env_get_home()));
+}
+
+int	go_to_path(const char *path)
+{
+	return (directory_change_dir(path));
+}
+
+int	builtin_cd(t_command *cmd)
+{
+	if (cmd->argv->argc == 1)
+		return(go_to_home());
+	else if (cmd->argv->argc == 2)
+	{
+		if (cmd->argv->argv[1][0] == '.')
+			return (go_to_path(cmd->argv->argv[1]));	/*check_for_real_path*/
+		else if (ft_strcmp(cmd->argv->argv[1], "-"))
+		{
+			int ret = go_to_prev_dir();
+			ft_dprintf(cmd->fds[FD_OUT], "%s\n", env_get_current_dir());
```

Comment:
cd never prints the directory

Reply:
In the man it says that :

 "-         When a <hyphen-minus> is used as the operand, this shall be equivalent to the command:
 97 
 98                      cd "$OLDPWD" && pwd
 99 
100                  which changes to the previous working directory and then writes its name."

Reply:
In the man it says
-         When a <hyphen-minus> is used as the operand, this shall be equivalent to the command:
 97 
 98                      cd "$OLDPWD" && pwd
 99 
100                  which changes to the previous working directory and then writes its name.

Reply:
In the man it says:
" -         When a <hyphen-minus> is used as the operand, this shall be equivalent to the command:
 97 
 98                      cd "$OLDPWD" && pwd
 99 
100                  which changes to the previous working directory and then writes its name.
"

Reply:
i stand corrected!